src/litserve/api.py

```diff
@@ -172,3 +174,11 @@ def encode_response(self, outputs):
                         yield encoded_output
              """
             )
+
+    def log(self, key, value):
+        """Log a key-value pair to the server."""
+        if self._logger_queue is None:
+            raise ValueError(
```

Comment:
this should not raise an error. 

you are allowed to call self.log without a logger. it will just be a no-op. instead, print a warning saying something like “you are recording self.log events but have no logger. if this is intended, then no action required. otherwise add a logger to track and visualize the logged metrics.